clientv3/ordering/kv.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ordering
+
+import (
+	"context"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/etcdserver/api/v3rpc/rpctypes"
+	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
+)
+
+type kvOrderingCache struct {
+	clientv3.KV
+	switchEndpoint SwitchEndpointClosure
```

Comment:
Alright that makes sense. Regardless of the how to switch endpoints and the retry rate implemented in the OrderViolationFunc type, the `Get` method should still contain the logic for retrying, right? So I was thinking of using: 
```
r, err := kv.KV.Do(ctx, clientv3.OpGet(key, opts...))
currRev := r.Get().Header.Revision
for err != nil && prevRev > currRev {
	kv.orderViolation(op, resp, prevRev)
	r, err = kv.KV.Do(ctx, clientv3.OpGet(key, opts...))
}
```

Does the above look right?